openwhisk/openwhisk.yaml

```diff
@@ -0,0 +1,100 @@
+# Task to define Runtime Parameters
+apiVersion: tekton.dev/v1alpha1
+kind: Task
+metadata:
+    name: openwhisk
+spec:
+    inputs:
+        resources:
+            - name: runtime-git
+              type: git
+        params:
+            - name: TARGET_IMAGE_NAME
+              description: name of the image to be tagged and pushed
+            - name: TARGET_IMAGE_TAG
+              description: tag the image before pushing
+              default: "latest"
+            - name: DOCKERFILE
+              description: The path to the dockerfile to build
+            - name: OW_RUNTIME_DEBUG
+              description: flag to indicate debug mode should be on/off
+              default: "false"
+            - name: OW_RUNTIME_PLATFORM
+              description: flag to indicate the platform, one of ["openwhisk", "knative", ... ]
+              default: "knative"
+            - name: OW_ACTION_NAME
+              description: name of the action
+              default: ""
+            - name: OW_ACTION_CODE
+              description: JavaScript source code to be evaluated
+              default: ""
+            - name: OW_ACTION_MAIN
+              description: name of the function in the "__OW_ACTION_CODE" to call as the action handler
+              default: "main"
+            - name: OW_ACTION_BINARY
+              description: flag to indicate zip function, for zip actions, "__OW_ACTION_CODE" must be base64 encoded string
+              default: "false"
+            - name: OW_HTTP_METHODS
+              description: list of HTTP methods, any combination of [GET, POST, PUT, and DELETE], default is [POST]
+              default: "[POST]"
+            - name: OW_ACTION_RAW
+              description: flag to indicate raw HTTP handling, interpret and process an incoming HTTP body directly
+              default: "false"
+            - name: OW_PROJECT_URL
+              description: Location to local or remote file storage or public or private GitHub repo from where JavaScript source code needs to be evaluated
```

Comment:
Local would be local to the git repo in the inputs, correct?
In case of remote, would it mean we have the docker file in a git repo and the action code somewhere else?
It seems that there is nothing JavaScript specific in the Task, should we mention the JavaScript runtime here?

Reply:
```OW_PROJECT_URL``` refers to the OpenWhisk action source, action source could be read directly from GitHub with raw source as explained in example here:

https://github.com/pritidesai/catalog/blob/openwhisk/openwhisk/runtimes/javascript/examples/03-github/taskrun-raw-github.yaml.tmpl#L50

Or, it could point to a zip file: 
https://github.com/pritidesai/catalog/blob/openwhisk/openwhisk/runtimes/javascript/examples/04-zip/taskrun-zip.yaml.tmpl#L52

Rephrased to remove reference to javascript from the description.